### PR TITLE
dnsdist: Use a counter to mark IDState usage instead of the FD

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

This change fixes a possible ABA issue and, coupled with stronger checks when updating the state, should reduce the likelihood of any issue when a state is reused from a different thread.

Needs a serious review.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [x] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
Travis failures are unrelated (network issues, apparently).

File: pdns/dnsdist.cc [hunk 1]
```diff
@@ -2091,19 +2090,15 @@ static void healthChecksThread()
       dss->prev.reuseds.store(dss->reuseds.load());
       
       for(IDState& ids  : dss->idStates) { // timeouts
-        int origFD = ids.origFD;
-        if(origFD >=0 && ids.age++ > g_udpTimeout) {
-          /* We set origFD to -1 as soon as possible
+        int64_t usageIndicator = ids.usageIndicator;
```
Comment (chbruyand) [inline pdns/dnsdist.cc]: Maybe use `.load()` ?

Comment (rgacogne) [reply pdns/dnsdist.cc]: Isn't that equivalent? Or do you think the use of `.load()` more readable, perhaps?

Comment (chbruyand) [reply pdns/dnsdist.cc]: It is. That was just to match https://github.com/PowerDNS/pdns/pull/8154/files#diff-41f3f8b6ff377ad8fb04d1a7a307a50eR547

**Pull request merged**